### ISSUE TITLE
s3.test.amazonaws.com and UnknownHostException

### ISSUE THREAD

**Issue by ksclarke** [Upplication/Amazon-S3-FileSystem-NIO2#57]
I've just started looking at this project so perhaps I'm doing something stupid. I've copied src/test/resources/amazon-test-sample.properties to amazon-test.properties and put in my credentials and bucket name. I can see that the credentials are good because I went from failing 43 tests to failing 40 (and I can use the credentials from the aws cli without issue).

All the unit tests succeed but when I run the integration tests, I get the same error for 40 of the tests:

    INFO  com.amazonaws.http.AmazonHttpClient  - Unable to execute HTTP request: ksclarketest.s3.test.amazonaws.com: unknown error
    java.net.UnknownHostException: ksclarketest.s3.test.amazonaws.com: unknown error

I can confirm http://ksclarketest.s3.test.amazonaws.com/ doesn't exist (DNS address could not be found) but http://ksclarketest.s3.amazonaws.com/ does (the bucket name is of course ksclarketest). I'm running:

    openjdk version "1.8.0_91"
    OpenJDK Runtime Environment (build 1.8.0_91-8u91-b14-0ubuntu4~16.04.1-b14)
    OpenJDK 64-Bit Server VM (build 25.91-b14, mixed mode)

on Ubuntu Gnome 16.04.

The README didn't indicate there were any other steps to running the integration tests other than copying the template config and putting valid values in it. Am I missing something? Still investigating, but wondering if it's just something simple that I'm missing. Thanks.

**Comment by ksclarke** [comment]
And, fwiw, the library itself does work for me. I can create a FileSystem and list the buckets on my S3 account.

**Comment by jarnaiz** [comment]
Hi Kevin,

First of all, thanks for your feedback :)

I think you are doing right and there isnt any other step to run the tests. I try to run the test without the amazon-test.properties and I have errors in 43 tests, like you.

I recently added jdk 8 to travis-ci.org and its working ok... Maybe its a permission problem with the access key and secret key. I will try to answer the issue #58 shortly but for now, Have you tried with all the permission?

You can try to add the access-key and secret-key by system environment var. Look at: ```com.upplication.s3fs.util.EnvironmentBuilder```for details

This are my reports without the amazon-test.properties and without system env vars:
```java

Tests in error: 
  putObjectWithBothSlash(com.upplication.s3fs.AmazonS3ClientIT)
  putObjectWithStartSlash(com.upplication.s3fs.AmazonS3ClientIT)
  putObject(com.upplication.s3fs.AmazonS3ClientIT)
  putObjectByteArray(com.upplication.s3fs.AmazonS3ClientIT)
  putObjectWithEndSlash(com.upplication.s3fs.AmazonS3ClientIT)
  createEmptyFileTest(com.upplication.s3fs.FilesOperationsIT)
  seekableCloseTwice(com.upplication.s3fs.FilesOperationsIT)
  amazonCopyDetectContentType(com.upplication.s3fs.FilesOperationsIT)
  virtualDirectoryStreamTest(com.upplication.s3fs.FilesOperationsIT)
  amazonOutpuStreamDetectContentType(com.upplication.s3fs.FilesOperationsIT)
  directoryStreamFirstDirTest(com.upplication.s3fs.FilesOperationsIT)
  buildEnv(com.upplication.s3fs.FilesOperationsIT)
  notExistsDir(com.upplication.s3fs.FilesOperationsIT)
  directoryStreamBaseBucketFindFileTest(com.upplication.s3fs.FilesOperationsIT)
  amazonCopyNotDetectContentTypeSetDefault(com.upplication.s3fs.FilesOperationsIT)
  copyUpload(com.upplication.s3fs.FilesOperationsIT)
  notExistsFile(com.upplication.s3fs.FilesOperationsIT)
  buildEnvAnotherURIReturnDifferent(com.upplication.s3fs.FilesOperationsIT)
  createEmptyDirTest(com.upplication.s3fs.FilesOperationsIT)
  copyDownload(com.upplication.s3fs.FilesOperationsIT)
  createFileWithFolderAndNotExistsFolders(com.upplication.s3fs.FilesOperationsIT)
  createTempFile(com.upplication.s3fs.FilesOperationsIT)
  deleteDir(com.upplication.s3fs.FilesOperationsIT)
  move(com.upplication.s3fs.FilesOperationsIT)
  existsFile(com.upplication.s3fs.FilesOperationsIT)
  directoryStreamBaseBucketFindDirectoryTest(com.upplication.s3fs.FilesOperationsIT)
  existsFileWithSpace(com.upplication.s3fs.FilesOperationsIT)
  readAttributesString(com.upplication.s3fs.FilesOperationsIT)
  deleteFullDirTest(com.upplication.s3fs.FilesOperationsIT)
  bucketIsDirectory(com.upplication.s3fs.FilesOperationsIT)
  createTempDir(com.upplication.s3fs.FilesOperationsIT)
  readAttributesDirectory(com.upplication.s3fs.FilesOperationsIT)
  virtualDirectoryStreamWithVirtualSubFolderTest(com.upplication.s3fs.FilesOperationsIT)
  copyDir(com.upplication.s3fs.FilesOperationsIT)
  readAttributesFile(com.upplication.s3fs.FilesOperationsIT)
  moveFromDifferentProviders(com.upplication.s3fs.FilesOperationsIT)
  deleteFile(com.upplication.s3fs.FilesOperationsIT)
  lookup_S3Object_when_S3Path_is_file_and_exists_other_starts_with_same_name(com.upplication.s3fs.S3UtilsIT)
  lookup_S3Object_when_S3Path_is_file(com.upplication.s3fs.S3UtilsIT)
  lookup_S3Object_when_S3Path_is_a_directory(com.upplication.s3fs.S3UtilsIT)
  lookup_S3Object_when_S3Path_is_a_directory_and_exists_other_directory_starts_same_name(com.upplication.s3fs.S3UtilsIT)
  lookup_S3Object_when_S3Path_is_a_directory_and_is_virtual(com.upplication.s3fs.S3UtilsIT)
  couldCreateFileSystem(com.upplication.s3fs.spike.EnvironmentIT)

Tests run: 50, Failures: 0, Errors: 43, Skipped: 0

````

Cheers

**Comment by ksclarke** [comment]
I did try with all permissions. With valid permissions the number of tests failed drops to 40 (from 43). The tests that fail still have the following exception:

    UnknownHostException: ksclarketest.s3.test.amazonaws.com

I wondered whether there might be something weird happening because I have 'test' in my bucket name. I'll keep investigating as time permits. Thanks.

**Comment by ksclarke** [comment]
I may be missing something, but I'm not sure how the integration tests are being run on Travis. When I look at your Travis config I see the env.Travis=true being set in a profile but I don't see how that triggers the integration tests. They don't run for me when I run `mvn -Denv.Travis=true clean install` on my machine but only when I run `mvn integration-test -Pintegration-tests` manually. Is that property doing something different on Travis than it is when I run it with my command line?

**Comment by jarnaiz** [comment]
Hi kevin,

I think i found the problem, I will try to solve the issue today. 
Its a problem with the ```S3UnitTestBase#S3_GLOBAL_URI```constant and my access key. If you change this to s3://s3.amazonaws.com/ you must solve the issue.

**Comment by ksclarke** [comment]
Yes, that was my theory too. I changed it to that but then one of the regular unit tests failed. I didn't investigate any further than that though. Glad you're on the trail of it, thanks!

**Comment by jarnaiz** [comment]
I think its solved in version 1.3.1 @ksclarke but let me know if something is still not working.
Thanks!

**Comment by ksclarke** [comment]
This issue is solved, but the integration tests still don't work for me. But, at least I'm getting different errors now. :-)

I'll file a different ticket for the new issue, if needed, once I dig into the new exceptions a bit more. Thanks!
